gorm/transaction.go

```diff
@@ -41,14 +41,14 @@ func FromContext(ctx context.Context) (txn *Transaction, ok bool) {
 // It works as a singleton to prevent an application of creating more than one
 // transaction instance per incoming request.
 type Transaction struct {
-	mu           sync.Mutex
-	parent       *gorm.DB
-	current      *gorm.DB
-	OnCommitHook func(ctx context.Context)
+	mu              sync.Mutex
+	parent          *gorm.DB
+	current         *gorm.DB
+	afterCommitHook []func(context.Context)
 }
 
-func (t *Transaction) SetOnCommitHook(f func(ctx context.Context)) {
-	t.OnCommitHook = f
+func (t *Transaction) AddAfterCommitHook(f []func(context.Context)) {
```

Comment:
the idea here would be that you add a single commit hook, and it appends it to the list:

```
func (t *Transaction) AddAfterCommitHook(hook func(context.Context)) {
    t.afterCommitHooks = append(t.afterCommitHooks, hook)
}
```

Better yet, you could make it variadic:
```
func (t *Transaction) AddAfterCommitHook(hooks ...func(context.Context)) {
    t.afterCommitHooks = append(t.afterCommitHooks, hooks...)
}
```

Reply:
you were perfectly clear btw, fault is all mine